Ошибки сборки clang'ом

Компилятор clang 13, собираю под FreeBSD, 2 ошибки:

1) Передаем в CExpc объект std::string вместо const char*
`morph_dict/AgramtabLib/agramtab_.cpp:163:90: error: cannot pass object of non-trivial type 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') through variadic function; call will abort at runtime [-Wnon-pod-varargs]
            throw CExpc(Format("line %s in  %s contains a dublicate gramcode", FileName, s));`

2) Не может отсортировать объекты CFuzzyResult из-за того что operator < не помечен как const

--- a/AgramtabLib/agramtab_.cpp
+++ b/AgramtabLib/agramtab_.cpp
@@ -160,7 +160,7 @@ void CAgramtab::Read(const char* FileName)
         size_t gram_index = GramcodeToLineIndex(s.c_str());
 
         if (GetLine(gram_index)) {
-            throw CExpc(Format("line %s in  %s contains a dublicate gramcode", FileName, s));
+            throw CExpc(Format("line %s in  %s contains a dublicate gramcode", FileName, s.c_str()));
         }
         GetLine(gram_index) = pAgramtabLine;
         if (!ProcessAgramtabLine(*this, s.c_str(), gram_index)) {

--- a/LemmatizerBaseLib/MorphAutomat.h
+++ b/LemmatizerBaseLib/MorphAutomat.h
@@ -118,7 +118,7 @@ public:
 struct CFuzzyResult {
 	std::string CorrectedString;
 	int StringDistance;
-	bool operator < (const  CFuzzyResult& x) {
+	bool operator < (const  CFuzzyResult& x) const {
 		return StringDistance < x.StringDistance;
 	}
 };

